src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -185,26 +172,35 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
         logger.trace("created new InternalEngine");
     }
 
+    private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boolean createNew) throws IOException {
+        final Translog.TranslogGeneration generation = loadTranslogIdFromCommit(writer);
+        Translog translog;
+        TranslogConfig translogConfig = engineConfig.getTranslogConfig();
+        if (createNew) {
+            translog = new Translog(translogConfig);
+        } else {
+            translogConfig.setTranslogGeneration(generation);
+            if (generation != null && generation.translogUUID == null) {
+                // only upgrade on pre-2.0 indices...
+                Translog.upgradeLegacyTranslog(logger, translogConfig);
+            }
+            translog = new Translog(translogConfig);
+        }
+
+        if (generation == null) {
+            logger.debug("no translog ID present in the current generation - creating one");
```

Comment:
This should only ever happen on a new index right?  Can we check that here, e.g. throw exc if `createNew` is false?